kernel/init.c

```diff
@@ -346,10 +348,11 @@ static void prepare_multithreading(struct k_thread *dummy_thread)
 	_kernel.ready_q.cache = &z_main_thread;
 #endif
 
-	z_setup_new_thread(&z_main_thread, z_main_stack,
-			   CONFIG_MAIN_STACK_SIZE, bg_thread_main,
-			   NULL, NULL, NULL,
-			   CONFIG_MAIN_THREAD_PRIORITY, K_ESSENTIAL, "main");
+	ret = z_setup_new_thread(&z_main_thread, z_main_stack,
+				 CONFIG_MAIN_STACK_SIZE, bg_thread_main,
```

Comment:
If i understand this correctly, this does not account for the guard area that in some cases is part of the thread stack memory allocation. Before this was not a problem because switch_to_main_thread was using the K_THREAD_STACK_SIZE_OF macro function, but now this is changed.

```
/*
 * Note:
 * The caller must guarantee that the stack_size passed here corresponds
 * to the amount of stack memory available for the thread.
 */
void z_setup_new_thread(struct k_thread *new_thread,

```

Reply:
you are right, the guard is not being accounted. Will close this pr, since @andrew is addressing it in a different pr though.